.github/workflows/nightly.yml

```diff
@@ -0,0 +1,117 @@
+# Copyright 2021 Agnostiq Inc.
+#
+# This file is part of Covalent.
+#
+# Licensed under the GNU Affero General Public License 3.0 (the "License").
+# A copy of the License may be obtained with this software package or at
+#
+#      https://www.gnu.org/licenses/agpl-3.0.en.html
+#
+# Use of this file is prohibited except in compliance with the License. Any
+# modifications or derivative works of this file must retain this copyright
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+name: nightly
+
+on:
+  schedule:
+    - cron: "0 4 * * *"
+
+jobs:
+  license:
+    name: License Scanner
+    uses: ./.github/workflows/license.yml@develop
+
+  tests:
+    name: Unit and Functional Tests
+    uses: ./.github/workflows/tests.yml@develop
+    secrets: inherit # pragma: allowlist secret
+
+  benchmarks:
+    name: Benchmarks
+    uses: ./.github/workflows/benchmark.yml@develop
+    secrets: inherit # pragma: allowlist secret
+
+  changelog:
+    name: Assign Version
+    needs:
+      - license
+      - tests
+    uses: ./.github/workflows/changelog.yml@develop
+    secrets: inherit # pragma: allowlist secret
+
+  push_to_master:
+    name: Push to Master
+    runs-on: ubuntu-latest
+    needs:
+      - changelog
+    steps:
+      - name: Get latest release
+        id: query-tags
+        uses: fjogeleit/http-request-action@v1
```

Comment:
any particular reason you're using this generic HTTP action instead of the official octokit action? probably doesn't matter, but I always use https://github.com/octokit/request-action

Reply:
wasnt' aware there existed an official version... happy to replace it with the better alternative